### PR TITLE
decouple dimension from split name

### PR DESCRIPTION
**Pull request by joesolly**
use the attribute name for looking up a dimension and return as that string instead of the internal druid dimension name

### CODE REVIEW

**Review by charlietanksley — APPROVED** [review]
This looks like a good change to me. Much more intuitive.

File: src/druidry/data_source.py [hunk 1]
```diff
@@ -324,11 +324,11 @@ def get_filter_for_dimension_choices(self, dimension, split, intervals=None, fil
                 ListFilter(dimension.dimension, split, choices))
 
     def get_split_exclusion_filter(self, splits, intervals=None, filter_=None):
-        dimensions = [(split, self.get_dimension(split)) for split in splits]
-        if dimensions:
+        if splits:
```
Comment (charlietanksley) [inline src/druidry/data_source.py:327]: 💯

File: src/druidry/data_source.py [hunk 2]
```diff
@@ -301,7 +296,11 @@ def get_metric(self, name):
         return next((m for m in self.metrics() if m.metric == name), None)
 
     def get_dimension(self, name):
-        return next((d for d in self.dimensions() if d.dimension == name), None)
+        dimension = getattr(self, name)
+        if isinstance(dimension, Dimension):
+            return dimension
+        else:
+            None
```
Comment (salman-monetate) [inline src/druidry/data_source.py:303]: Does it have to return None?

Comment (joesolly) [reply src/druidry/data_source.py:303]: i like it being explicit, the other option is making it raise an exception

**Review by salman-monetate — APPROVED** [review]

**Pull request merged**